src/daft-local-execution/src/lib.rs

```diff
@@ -6,14 +6,99 @@ mod run;
 mod runtime_stats;
 mod sinks;
 mod sources;
+use std::{
+    future::Future,
+    panic::AssertUnwindSafe,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, OnceLock,
+    },
+};
+
 use common_error::{DaftError, DaftResult};
+use futures::FutureExt;
 use lazy_static::lazy_static;
 pub use run::NativeExecutor;
 use snafu::{futures::TryFutureExt, Snafu};
 lazy_static! {
     pub static ref NUM_CPUS: usize = std::thread::available_parallelism().unwrap().get();
 }
 
+static COMPUTE_RUNTIME: OnceLock<RuntimeRef> = OnceLock::new();
+
+pub type RuntimeRef = Arc<Runtime>;
+
+pub struct Runtime {
```

Comment:
as discussed offline, move to common crate and have the `static` runtimes for io and compute defined there